src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,116 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_SIZE];
+    }
 
     @Override
     public void add(T value) {
-
+        if (array.length == size) {
+            array = grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException("Add element failed. Index out of bounds");
+        } else {
+            if (size == array.length) {
+                array = grow();
+            }
+            for (int i = size - 1; i >= index; i--) {
+                moveRight(i);
+            }
+            array[index] = value;
+            size++;
+        }
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int listSize = list.size();
+        for (int i = 0; i < listSize; i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < size && index >= 0) {
+            return array[index];
+        }
+        throw new ArrayIndexOutOfBoundsException("Getting element failed. Index out of bounds");
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < size && index >= 0) {
```

Comment:
let's make a boolean method(checkIndex) and use it in set(), remove(), get()

Reply:
And let's use this order everywhere:
if(condition) {throw new Exception}
//rest part of the method